recovery/index.html

```diff
@@ -262,6 +262,64 @@ <h2>Message log</h2>
         return bytes;
       }
 
+      /**
+       * Decodes a base58check-encoded string into a buffer
+       * This function throws an error when the string is too small, doesn't have the proper checksum, or contains invalid characters.
+       * Inspired by https://gist.github.com/diafygi/90a3e80ca1c2793220e5/
+       * @param {string} s
+       * @return {Uint8Array}
+       */
+      async function base58checkDecode(s) {
+        if (s.length < 5) {
+          throw new Error(`cannot base58-decode a string of length < 5 (found length ${s.length})`)
+        }
+
+        // See https://en.bitcoin.it/wiki/Base58Check_encoding
+        var alphabet = "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
+        var decoded = BigInt(0);
+        var decodedBytes = [];
+        var result = [];
+        for (var i = 0; i < s.length; i++) {
+          if (alphabet.indexOf(s[i]) === -1) {
+            throw new Error(`cannot base58-decode: ${s[i]} isn't a valid character`)
+          }
+          var carry = alphabet.indexOf(s[i]);
+
+          // If the current base58 digit is 0, append a 0 byte.
+          // "result.length == i" can only happen if we have not seen non-zero bytes so far
```

Comment:
don't follow this second line. Reads to me like `result.length == i` will only be true if you pushed non 0-bits. But I think you mean it means that you have seen at least 1 byte

Reply:
`result` actually contains _only_ zero bytes at this stage. So, if we had a non-zero bytes that we iterated over, result is untouched which means its length doesn't change. But `i` keeps increasing. I can change the check to result `i > result.length`. And maybe rename result to `leadingZeros`?

Reply:
Ok done in the latest diff, let me know what you think!